From 8a495ce8998de05db5a7e3d67a9ef0124f939f5f Mon Sep 17 00:00:00 2001
From: Nikhil Badola <nikhil.badola@freescale.com>
Date: Tue, 20 May 2014 16:54:02 +0530
Subject: [PATCH 1071/1094] drivers/usb : Add affected configs for Erratum
 A005697 workaround

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Add following affected SOCs and their variants
	P1010 rev2.0
	B4860 rev1.0 rev2.0 rev2.1 rev2.2
	T2080 rev1.0
	T2081 rev1.0

Signed-off-by: Nikhil Badola <nikhil.badola@freescale.com>
Change-Id: I954877e5b878abf63667b0215aa41a8a3a93eddf
Reviewed-on: http://git.am.freescale.net:8181/12743
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
---
 drivers/usb/host/fsl-mph-dr-of.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/fsl-mph-dr-of.c b/drivers/usb/host/fsl-mph-dr-of.c
index a765710..f4b0537 100644
--- a/drivers/usb/host/fsl-mph-dr-of.c
+++ b/drivers/usb/host/fsl-mph-dr-of.c
@@ -150,9 +150,10 @@ static bool has_erratum_a005697(void)
 	bool flag = false;
 
 	switch (SVR_SOC_VER(svr)) {
-	case SVR_P1010:
 	case SVR_P1014:
 	case SVR_T1040:
+	case SVR_T2080:
+	case SVR_T2081:
 		if (SVR_REV(svr) == 0x10)
 			flag = true;
 		break;
@@ -166,6 +167,7 @@ static bool has_erratum_a005697(void)
 				(SVR_REV(svr) == 0x21))
 			flag = true;
 		break;
+	case SVR_P1010:
 	case SVR_T4240:
 	case SVR_T4160:
 	case SVR_P5020:
@@ -185,6 +187,12 @@ static bool has_erratum_a005697(void)
 				(SVR_REV(svr) == 0x30))
 			flag = true;
 		break;
+	case SVR_B4860:
+	case SVR_B4420:
+		if ((SVR_REV(svr) == 0x10) || (SVR_REV(svr) == 0x20) ||
+		    (SVR_REV(svr) == 0x21) || (SVR_REV(svr) == 0x22))
+			flag = true;
+		break;
 	}
 
 	return flag;
-- 
2.0.2

